builds/e2e/templates/e2e-run-base-image-update.yaml

```diff
@@ -0,0 +1,139 @@
+# This file is a copy of e2e-run.yaml.
+# The *only* difference is that a different E2E_REGISTRIES__0__PASSWORD is being set here.
+# Passing variables across templates was proving tricky and needs some looking into. However,
```

Comment:
What exactly do you mean when you say "passing variables across templates"? Does the following link help at all? https://github.com/MicrosoftDocs/azure-devops-docs/issues/4543

Reply:
Possibly. I'm playing with it now.

But yes, the essence was passing a variable (or in this case a secret) from one template to another. It took a bit before I realized that it's not a one size fits all; the steps for passing a variable across tasks or stages is different to passing one across templates.